clients/drcachesim/analyzer.cpp

```diff
@@ -443,7 +493,9 @@ analyzer_tmpl_t<RecordType, ReaderType>::process_tasks(analyzer_worker_data_t *w
             }
             return;
         }
-        int shard_index = worker->stream->get_input_stream_ordinal();
+        int shard_index = shard_type_ == SHARD_BY_CORE
+            ? worker->index
```

Comment:
We're tying workers to cores. What if I can have only X workers but have Y core traces to process, with X < Y. Like the thread shard case, we should be able to map multiple core shards to the same worker.

Reply:
I don't think that is a good idea.  It is going to cause a multiplexed core to lag behind the others and result in a skewed schedule.  What is the reason you have limited workers?

Reply:
> What is the reason you have limited workers?

What I mean is, these are software threads: no matter how many physical cores you have for parallelism, you can still have #cores threads and let the OS multiplex them which will be a fair scheduling.  I don't see any advantage to  us trying to implement some fair multiplexing ourselves: it seems like it will take development time for no benefit.

Reply:
> let the OS multiplex them which will be a fair scheduling

Maybe it's worth highlighting that results from some of these flags are system-dependent.

> I don't see any advantage to us trying to implement some fair multiplexing ourselves: it seems like it will take development time for no benefit.

It will help get consistent results across runs done on different machines. But I agree we shouldn't go do that right now.

I assume when you say a "fair multiplexing" scheme, you also mean a schedule where we *statically* map portions of each thread's trace to cores.

Reply:
> I assume when you say a "fair multiplexing" scheme, you also mean a schedule where we statically map portions of each thread's trace to cores.

No, if we supported #workers != #cores we'd have to handle dynamic schedules, not just static.  We'd have to run one core output stream for a while and then switch to another, with some quantum definition: a whole new scheduler just to support that mode, which is why it doesn't seem worth all that effort.